### ISSUE TITLE
div-icon helper with component?

### ISSUE THREAD

**Issue by localyost** [miguelcobain/ember-leaflet#79]
Hi guys,

I've been looking at issue #33 and am probably missing something. 
I'm trying to use the div-icon helper to render a component like this:

`{{marker-layer icon=(div-icon html=(component 'icon-component')) location=markerLocation}}`

`icon-component` is a component that should be rendered as html all I'm getting is the object (`[object Object]`) and `init()` is not being called. Any Idea? Thank you!

**Comment by miguelcobain** [comment]
@localyost what you want to do is impossible.

1. `(component 'icon-component')` returns a component instance, not an HTML string or pure DOM
2. even if it returned pure DOM, it still wouldn't work because leaflet div icons only accepts HTML strings

tl;dr is that we need to pass in a string **OR** a helper that returns a string.
If you want to encapsulate the string, you have to use a property coming from the template's context or a helper that returns that string.
Only strings for now. :(

There's an open issue on leaflet repo for this.

**Comment by localyost** [comment]
ok, thanks!
